zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/ZNodeGroupAclUtil.java

```diff
@@ -33,4 +38,39 @@
   // Has the same effect as the ZK client using ZooDefs.Ids.CREATOR_ALL_ACL.
   public static final String SET_X509_CLIENT_ID_AS_ACL =
       ZNODE_GROUP_ACL_CONFIG_PREFIX + "setX509ClientIdAsAcl";
+  // A list of domain names that will have super user privilege, separated by ","
+  public static final String SUPER_USER_DOMAIN_NAME =
+      ZNODE_GROUP_ACL_CONFIG_PREFIX + "superUserDomainName";
+  // A list of znode path prefixes, separated by ","
+  // Znode whose path starts with the defined path prefix would have open read access
+  // Meaning the znode will have (world:anyone, r) ACL
+  public static final String OPEN_READ_ACCESS_PATH_PREFIX =
+      ZNODE_GROUP_ACL_CONFIG_PREFIX + "openReadAccessPathPrefix";
+
+  /**
+   * Get open read access path prefixes from config
+   * @return A set of path prefixes
+   */
+  public static Set<String> getOpenReadAccessPathPrefixes() {
+    String openReadAccessPathPrefixesStr = System.getProperty(OPEN_READ_ACCESS_PATH_PREFIX);
+    if (openReadAccessPathPrefixesStr == null || openReadAccessPathPrefixesStr.isEmpty()) {
+      return Collections.emptySet();
+    }
+    return Arrays.stream(openReadAccessPathPrefixesStr.split(",")).filter(str -> str.length() > 0)
+        .collect(Collectors.toSet());
+  }
+
+  /**
+   * Get the domain names that are mapped to super user access privilege
+   * @return A set of domain names
+   */
+  public static Set<String> getSuperUserDomainNames() {
+    String superUserDomainNameStr =
+        System.getProperty(ZNodeGroupAclUtil.SUPER_USER_DOMAIN_NAME);
+    if (superUserDomainNameStr == null || superUserDomainNameStr.isEmpty()) {
+      return Collections.emptySet();
+    }
+    return Arrays.stream(superUserDomainNameStr.split(",")).filter(str -> str.length() > 0)
+        .collect(Collectors.toSet());
+  }
```

Comment:
Another comment - I think we should make this an object and assign some state to it. By that I mean that it's pretty inefficient to read a string and split it every time a request gets processed in PrepRequestProcessor. Do you think we could update methods in this class to cut down on repetitive work?